holoviews/ipython/display_hooks.py

```diff
@@ -146,7 +143,13 @@ def display_figure(fig, message=None, max_width='100%'):
     dpi = OutputMagic.options['dpi']
     backend = OutputMagic.options['backend']
 
-    if backend == 'd3' and mpld3:
+    if backend == 'nbagg':
+        import numpy as np
+        from matplotlib.backends.backend_nbagg import FigureManagerNbAgg, new_figure_manager_given_figure
```

Comment:
I would move this to a conditional import at the top of the file and then check for None here. E.g if ``FigureManagerNbAgg` is None`` (because the import failed) you could then generate an appropriate message here.